packages/client/src/workers/index.ts

```diff
@@ -0,0 +1,37 @@
+import { deserialize, serialize } from '../serializer';
+import { Request } from '../util/request';
+
+type XataWorkerContext<XataClient> = { xata: XataClient; request: Request; env: Record<string, string | undefined> };
+
+type RemoveFirst<T> = T extends [any, ...infer U] ? U : never;
+
+type WorkerRunnerConfig = {
+  workspace: string;
+  worker: string;
+};
+
+export function buildWorkerRunner<XataClient>(config: WorkerRunnerConfig) {
+  return function xataWorker<WorkerFunction extends (ctx: XataWorkerContext<XataClient>, ...args: any[]) => any>(
+    name: string,
+    _worker: WorkerFunction
+  ) {
+    return async (...args: RemoveFirst<Parameters<WorkerFunction>>): Promise<Awaited<ReturnType<typeof _worker>>> => {
+      const url =
+        process.env.NODE_ENV === 'development'
+          ? `http://localhost:64749/${name}`
```

Comment:
I guess this won't work if the user specifies a different port in the CLI, right?

Reply:
Yeah, I'll remove the customization of the port for now and we revisit it